sklearn/model_selection/validate.py

```diff
@@ -0,0 +1,905 @@
+"""
+The :mod:`sklearn.model_selection.validate` module includes classes and
+functions to validate the model.
+"""
+
+# Author: Alexandre Gramfort <alexandre.gramfort@inria.fr>,
+#         Gael Varoquaux <gael.varoquaux@normalesup.org>,
+#         Olivier Girsel <olivier.grisel@ensta.org>
+# License: BSD 3 clause
+
+
+from __future__ import print_function
+from __future__ import division
+
+import warnings
+import numbers
+import time
+
+import numpy as np
+import scipy.sparse as sp
+
+from ..base import is_classifier, clone
+from ..utils import indexable, check_random_state, safe_indexing
+from ..utils.fixes import astype
+from ..utils.validation import _is_arraylike, _num_samples
+from ..externals.joblib import Parallel, delayed, logger
+from ..metrics.scorer import check_scoring
+from .split import check_cv, _safe_split
+
+
+__all__ = ['cross_val_score', 'cross_val_predict', 'permutation_test_score',
+           'learning_curve', 'validation_curve']
+
+
+# TODO Move this into sklearn.exceptions
+class FitFailedWarning(RuntimeWarning):
+    pass
+
+
+def cross_val_score(estimator, X, y=None, labels=None, scoring=None, cv=None,
```

Comment:
So what do we do?? do we keep the `labels`'s order as before (if exists) and use `X, y, labels` if `labels` did not previously exist...